test/e2e/gas-price.feature.ts

```diff
@@ -0,0 +1,147 @@
+import { ethers } from 'ethers';
+import * as hre from 'hardhat';
+
+import '@nomiclabs/hardhat-ethers';
+import {
+  airseekerV2ProviderRecommendedGasPrice,
+  multiplyGasPrice,
+  gasPriceStore,
+  initializeGasStore,
+  clearExpiredStoreGasPrices,
+} from '../../src/gas-price/gas-price';
+
+const chainId = '31337';
+const providerName = 'localhost';
+const rpcUrl = 'http://127.0.0.1:8545/';
+const gasSettings = {
+  recommendedGasPriceMultiplier: 1.5,
+  sanitizationSamplingWindow: 15,
+  sanitizationPercentile: 80,
+  scalingWindow: 2,
+  scalingMultiplier: 2,
+};
+const provider = new hre.ethers.providers.StaticJsonRpcProvider(rpcUrl);
+const timestampMock = 1_696_930_907_351;
+
+const sendTransaction = async (gasPriceOverride?: ethers.BigNumber) => {
+  const wallets = await hre.ethers.getSigners();
+  const wallet = wallets[0]!;
+
+  await wallet.sendTransaction({
+    to: hre.ethers.constants.AddressZero,
+    ...(gasPriceOverride ? { gasPrice: gasPriceOverride } : {}),
+  });
+};
+
+describe('airseekerV2ProviderRecommendedGasPrice', () => {
+  beforeEach(async () => {
+    // Reset the local hardhat network state for each test to prevent issues with other test contracts
+    await hre.network.provider.send('hardhat_reset');
```

Comment:
It's useful to reset the provider recommended gas price since otherwise it would fluctuate between tests. We also have the [runInBand](https://jestjs.io/docs/cli#--runinband) option set to run them sequentially.

Reply:
Thanks. I forgot we added that in the other PR.